src/validators/validation_state.rs

```diff
@@ -61,6 +64,18 @@ impl<'a, 'py> ValidationState<'a, 'py> {
             Some(Exactness::Exact) => self.exactness = Some(exactness),
         }
     }
+
+    pub fn cache_str(&self) -> bool {
+        self.extra.cache_str
+    }
+
+    pub fn maybe_cached_str(&self, py: Python<'py>, s: &str) -> Bound<'py, PyString> {
+        if self.cache_str() {
+            jiter::cached_py_string(py, s)
+        } else {
+            PyString::new_bound(py, s)
+        }
+    }
```

Comment:
Probably for these functions for now we should only cache if user says cache everything (which is the default anyway), just to be conservative. We can always thread through the key context later, perhaps.